src/interface/pipe.rs

```diff
@@ -49,18 +91,41 @@ impl EmulatedPipe {
         }
     }
 
+    /// # Description
+    /// Setter for EOF flag
     pub fn set_eof(&self) {
         self.eof.store(true, Ordering::Relaxed);
     }
 
+    /// # Description
+    /// Getter for write references
+    ///
+    /// # Returns
+    ///
+    /// Number of references to write end of the pipe
+    ///
```

Comment:
Why do you need this?  In what circumstances?  

(I have similar comments below for many other functions)